src/pages/selectApptToConfirm.tsx

```diff
@@ -0,0 +1,56 @@
+import AppointmentCard from "@/components/appointment/AppointmentCard";
+import Navbar from "@/components/public/Navbar";
+import { Appointment } from "@/types/Appointment";
+import { Typography, Grid, Box } from "@mui/material";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useContext, useEffect, useState } from "react";
+import { Database } from "supabase/db_types";
+import { userContext } from "supabase/user_context";
+import { NextRouter, useRouter } from "next/router";
+import { PagePaths } from "enum/pages";
+import { GetAppointmentsByUserIdWhichPending } from "@/services/Appointments";
+
+export default function Home() {
+  const router: NextRouter = useRouter();
+
+  const supabaseClient = useSupabaseClient<Database>();
+  const [appointments, setAppointments] = useState<Appointment[]>([]);
+  const userStatus = useContext(userContext);
+
+  function handleCardClick(appointmentId: string): void {
+    router.push(PagePaths.confirmAppt + appointmentId);
+  }
+
+  useEffect(() => {
+    if (!userStatus.user) return;
+
+    GetAppointmentsByUserIdWhichPending(userStatus.user.userId, supabaseClient)
+      .then((appointment) => {
+        setAppointments(appointment);
+      })
+      .catch((err) => {
+        console.log(err);
+        return;
+      });
+  }, [supabaseClient, userStatus.user]);
+
+  return (
```

Comment:
Anyone can access this page (including without login🤨🤨)

Reply:
guard added. now unlogged in user can't access this page